pandas/io/sql.py

```diff
@@ -295,6 +301,7 @@ def read_sql_query(
     params=None,
     parse_dates=None,
     chunksize: None = None,
+    dtype: Optional[Union[Dtype, Dict[str, Dtype]]] = None,
```

Comment:
this type actually is pretty usefule, can you define in _typing, call it DtypeOrDictDtype / DtypeTable and add a comment about it. cc @simonjayhawkins @WillAyd @jorisvandenbossche for the name here.

Reply:
```suggestion
    dtype: Optional[Union[Dtype, Dict[Label, Dtype]]] = None,
```

Maybe `DtypeArg`?

Reply:
Done, added `DtypeArg` to _typing